include/mbedtls/ssl_internal.h

```diff
@@ -705,22 +751,26 @@ void mbedtls_ssl_read_version( int *major, int *minor, int transport,
 
 static inline size_t mbedtls_ssl_hdr_len( const mbedtls_ssl_context *ssl )
 {
+#if !defined(MBEDTLS_SSL_PROTO_DTLS) || !defined(MBEDTLS_SSL_PROTO_TLS)
+    ((void) ssl);
+#endif
+
 #if defined(MBEDTLS_SSL_PROTO_DTLS)
-    if( ssl->conf->transport == MBEDTLS_SSL_TRANSPORT_DATAGRAM )
+    MBEDTLS_SSL_TRANSPORT_IF_DTLS( ssl->conf->transport )
         return( 13 );
-#else
-    ((void) ssl);
 #endif
     return( 5 );
 }
 
 static inline size_t mbedtls_ssl_hs_hdr_len( const mbedtls_ssl_context *ssl )
 {
+#if !defined(MBEDTLS_SSL_PROTO_DTLS) || !defined(MBEDTLS_SSL_PROTO_TLS)
```

Comment:
I expect this will occur many times - shall we introduce `MBEDTLS_SSL_PROTO_BOTH` as an abbreviation?